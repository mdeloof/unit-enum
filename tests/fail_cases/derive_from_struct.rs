use unit_enum::UnitEnum;

#[derive(UnitEnum)]
struct Vec2 {
    x: i32,
    y: i32,
}

fn main() {}
