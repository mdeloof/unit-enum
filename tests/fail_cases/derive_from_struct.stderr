error: item is not an enum

         = help: `ItemEnum` can only be derived from an enum

 --> tests/fail_cases/derive_from_struct.rs:4:1
  |
4 | / struct Vec2 {
5 | |     x: i32,
6 | |     y: i32,
7 | | }
  | |_^
